src/main/java/org/openmrs/maven/plugins/utility/ModuleManager.java

```diff
@@ -0,0 +1,125 @@
+package org.openmrs.maven.plugins.utility;
+
+import org.apache.maven.artifact.versioning.ArtifactVersion;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+import org.openmrs.maven.plugins.ModuleInstall;
+import org.openmrs.maven.plugins.model.Artifact;
+import org.openmrs.maven.plugins.model.Server;
+import org.twdata.maven.mojoexecutor.MojoExecutor;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.twdata.maven.mojoexecutor.MojoExecutor.*;
+
+/**
+ * Handles installing modules on server
+ */
+public class ModuleManager {
+
+    private static final String GOAL_UNPACK = "unpack";
+
+    MavenProject mavenProject;
+
+    MavenSession mavenSession;
+
+    BuildPluginManager pluginManager;
+
+    VersionsHelper versionsHelper;
+
+    public ModuleManager(MavenProject mavenProject,
```

Comment:
Add default constructor and annotate fields with @Requirement so it can be created by DI container.